### ISSUE TITLE
go-doc-browser behaves different than go-doc

### ISSUE THREAD

**Issue by josefcs** [fatih/vim-go#1058]
Given the following example:

```
package main

import "net/http"

func main() {
	srv := &http.Server{}
	srv.ListenAndServe()
}
```

### Actual behavior

When opening the go-doc on `ListenAndServe`, the correct documentation for `net/http.Server` comes up.
But when opening it in go-doc-browser, go-vim tries to open `https://godoc.org/srv#ListenAndServe`, which of course can't be found.

### Expected behavior

In this case, go-doc-browser should open `https://godoc.org/net/http#ListenAndServe` (like go-doc).

**Comment by fatih** [comment]
Hi @josefcs 

That's expected because `go-doc` uses `getgodoc`. But `go-doc-browser` just uses an internal parser, and it's not that smart. I'll try to find change it so it uses `getgodoc` if possible. Thanks for the report

**Comment by fatih** [comment]
This is fixed now with the `vim-8.0` branch. It requires `json_encode`. Thanks for the report again.

**Comment by kaihendry** [comment]
I'm running into this issue with go-doc-browser on 08c881dea2eee7ef208a8f27131b6696478bfb99 in VIM - Vi IMproved 8.0 on Archlinux. Is this is a new issue?
